app/helpers/sessions_helper.rb

```diff
@@ -3,15 +3,36 @@ def log_in user
     session[:user_id] = user.id
   end
 
+  def remember(user)
+    user.remember
+    cookies.permanent.signed[:user_id] = user.id
```

Comment:
signed để làm gì e ? có cần thiết k e ?

Reply:
Với cookie thông thường được đặt, không có tùy chọn ở phía ứng dụng để tìm hiểu xem người dùng ứng dụng đã thay đổi cookie. Điều này dẫn đến các vấn đề bảo mật, vì việc thay đổi nội dung của một cookie trong trình duyệt không phải là khó. Giải pháp là đăng ký các cookie với một key chỉ được biết với chúng ta, key này được tạo tự động. Sử dụng phương thức signed để viết và đọc cookie. Em đọc cũng chưa hiểu rõ lắm, với thời điểm hiện tại em nghĩ nó chưa cần thiết anh

Reply:
Ngắn gọn hơn theo em hiểu, phương thức signed sẽ mã hóa cookie để đảm bảo an toàn bảo mật ạ